lookout/core/sqla_model_repository.py

```diff
@@ -66,11 +67,22 @@ def __init__(self, db_endpoint: str, fs_root: str, max_cache_mem: int, ttl: int,
         :param engine_kwargs: Passed directly to SQLAlchemy's `create_engine()`.
         """
         self.fs_root = fs_root
+        db_endpoint_components = urlparse(db_endpoint)
+        if db_endpoint_components.password is not None:
+            password, netloc = db_endpoint_components.password, db_endpoint_components.netloc
+            password_index = netloc.rindex(password)
+            safe_netloc = "%s%s%s" % (
+                db_endpoint_components.netloc[:password_index],
+                "<PASSWORD>",
+                db_endpoint_components.netloc[password_index + len(password):])
+            safe_db_endpoint_components = list(db_endpoint_components)
+            safe_db_endpoint_components[1] = safe_netloc
+            self.safe_db_endpoint = urlunparse(safe_db_endpoint_components)
```

Comment:
Let's make it `_safe_db_endpoint` and a readonly property.

Reply:
I added the `_`.

For the readonly part, do you have a sensible pattern to make a private attribute readonly in mind?

```python
class A:
    def __init__(self):
        self._a_value = 1

    @property
    def _a(self):
        return self._a_value
```
This one for example doesn't make much sense since if someone wants to edit `_a`, `_a_value` is just as accessible. I propose not to bother and just use `_` but if you have another pattern in mind I'll push it :)

Reply:
Sorry for the confusion, I simply meant
```python
def safe_db_endpoint(self):
    return self._safe_db_endpoint
```

But anyway, we don't need it anywhere so good to merge as-is